assets/src/edit-story/elements/text/display.js

```diff
@@ -164,30 +171,50 @@ function TextDisplay({
       ? `${dataToEditorY(-updatedMargin) / 2}px 0`
       : '';
 
-    if (outerBorderRef.current) {
+    if (outerBorderRef.current || bgRef.current) {
+      // Depending on the background mode, choose the element that has border assigned to it.
       if (transform) {
         const { resize } = transform;
         if (resize && resize[0] !== 0 && resize[1] !== 0) {
           const [width, height] = resize;
           if (isOutsideBorder(border)) {
-            outerBorderRef.current.style.width =
+            refWithBorder.current.style.width =
               width + border.left + border.right + 'px';
-            outerBorderRef.current.style.height =
+            refWithBorder.current.style.height =
               height + border.top + border.bottom + 'px';
           }
         }
       }
     }
   });
 
+  const innerElementTracker = useRef(null);
+  useEffect(() => {
+    if (fgRef.current?.children) {
+      innerElementTracker.current = fgRef.current.children[0];
```

Comment:
Unfortunately, the first child here is only the first line of text, so if you make multiple lines of text (using actual newlines, not just line-wrap), this will make only the first line update instantly whereas subsequent lines are updated with a delay.

So either you have to use a higher-level node (which I guess you haven't because it didn't work) or you could make your transform handler work also if the ref is pointing to a `NodeList` and not just a single node - and simply point it to `children` here:

```js
      innerElementTracker.current = fgRef.current.children;
```

Or use a CSS variable, as mentioned in the next comment.

Reply:
Added in 3e0a3f1 without CSS variables -- didn't seem that straightforward to use CSS var on each child. Looping through the children instead to assign the style.

Reply:
Awesome - works great!